windows-containers/productization.md

```diff
@@ -0,0 +1,180 @@
+
+### Implementation Details
+
+#### Windows Scale Up (WSU)
+
+The Windows Scale Up is an Ansible playbook that has the follow prerequisites:
+* Needs to run on a Linux system
+* Needs to be able to access the cluster where the Windows worker will be added
+* Needs to be able to access the Windows node
+
+The inputs that the WSU playbook requires are:
+* Windows node credentials
+* Kubelet download location
+* Worker ignition
+* Network configuration (?)
+  * The details of this will be worked out as part of the enhancement
+    proposal for "OVN Plumbing for Hybrid Linux+Windows Clusters GA"
+* WMCB download location
+
+The actions that the WSU will perform are:
+* Check if the container runtime is present on the Windows node
+* Download and copy the kubelet to the Windows node
+* Extract the worker ignition from the MCS pods and copy it to the Windows node
+  * This is an option only for user provisioned clusters. We are investigating
+    alternatives for installer provisioned clusters.
+* Extract and copy the network configuration to the Windows node (?)
+* Download the WMCB, copy it to the Windows node and execute it
+
+#### Windows Machine Config Bootstrapper (WMCB)
+
+The Windows Machine Config Bootstrapper executable has the following
+prerequisites:
+* Only supports Windows x86-64 nodes (Windows Server 2019)
+* Can only run on Windows nodes
+
+The inputs that the WMCB requires are:
+* kubelet location on the local disk
+* Worker ignition location on the local disk
+* Network configuration location on the local disk (?)
+  * The details of this will be worked out as part of the enhancement
+    proposal for "OVN Plumbing for Hybrid Linux+Windows Clusters GA"
+
+The actions that the WMCB will perform are:
+* Install and configure kubelet
+* Parse the worker ignition and extract the bootstrap kubeconfig and the
+kubelet configuration
+  * This is the only use case for the ignition file
+* Launch the kubelet as a Windows service
+* Check if the kubelet is running
+* Exit
+
+### Risks and Mitigations
+
+The main risk with this proposal is the dependency on Microsoft to publish a
+downstream version of the kubelet. If this does not happen we would have to
+use the upstream version of the kubelet which will result in Red Hat being
+responsible for security and other fixes.
+
+## Design Details
+
+### Test Plan
+
+We plan to have all the repositories associated with this effort fully
+integrated with Prow CI and run e2e tests for every PR that is opened. These
+e2e tests will involve bringing up a cluster on all supported cloud providers,
+instantiating a Windows node and running workloads on it. We also plan to add
+blocking tests to the nightly runs.
+
+### Graduation Criteria
+
+This enhancement will start as GA
+
+### Upgrade / Downgrade Strategy
+
+We will support upgrades of the node components by publishing a new version
+of the WSU Ansible playbook. An older version of the playbook can be used to
```

Comment:
You mean a `older release`?

Reply:
Don't they mean the same thing? I can replace `version` with `release` in both cases if you prefer that.